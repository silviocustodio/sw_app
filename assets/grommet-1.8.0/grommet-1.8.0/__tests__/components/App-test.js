// (C) Copyright 2014-2016 Hewlett Packard Enterprise Development LP

import React from 'react';
import renderer from 'react-test-renderer';

import App from '../../src/js/components/App';

// needed because this:
// https://github.com/facebook/jest/issues/1353
jest.mock('react-dom');

describe('App', () => {
  it('has correct default options', () => {
    const component = renderer.create(
      <App><h2>App Body</h2></App>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('has correct inline=true rendering', () => {
    const component = renderer.create(
      <App inline={true} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('has correct lang="pt=BR" rendering', () => {
    const component = renderer.create(
      <App lang='pt-BR' />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('has correct className rendering', () => {
    const component = renderer.create(
      <App className='testing' />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('has microdata properties rendering', () => {
    const component = renderer.create(
      <App itemScope={true} itemType="http://schema.org/Article"
        itemProp="test"/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
