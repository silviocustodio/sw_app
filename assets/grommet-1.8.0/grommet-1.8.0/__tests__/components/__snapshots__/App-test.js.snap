// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App has correct className rendering 1`] = `
<div
  className="grommet grommetux-app grommetux-app--centered testing"
  lang="en-US"
>
  <span
    style={
      Object {
        "display": "none",
      }
    }
  />
  <div
    aria-live="polite"
    className="grommetux-app__announcer"
  />
</div>
`;

exports[`App has correct default options 1`] = `
<div
  className="grommet grommetux-app grommetux-app--centered"
  lang="en-US"
>
  <h2>
    App Body
  </h2>
  <span
    style={
      Object {
        "display": "none",
      }
    }
  />
  <div
    aria-live="polite"
    className="grommetux-app__announcer"
  />
</div>
`;

exports[`App has correct inline=true rendering 1`] = `
<div
  className="grommet grommetux-app grommetux-app--centered grommetux-app--inline"
  lang="en-US"
>
  <span
    style={
      Object {
        "display": "none",
      }
    }
  />
  <div
    aria-live="polite"
    className="grommetux-app__announcer"
  />
</div>
`;

exports[`App has correct lang="pt=BR" rendering 1`] = `
<div
  className="grommet grommetux-app grommetux-app--centered"
  lang="pt-BR"
>
  <span
    style={
      Object {
        "display": "none",
      }
    }
  />
  <div
    aria-live="polite"
    className="grommetux-app__announcer"
  />
</div>
`;

exports[`App has microdata properties rendering 1`] = `
<div
  className="grommet grommetux-app grommetux-app--centered"
  itemProp="test"
  itemScope={true}
  itemType="http://schema.org/Article"
  lang="en-US"
>
  <span
    style={
      Object {
        "display": "none",
      }
    }
  />
  <div
    aria-live="polite"
    className="grommetux-app__announcer"
  />
</div>
`;
